h/static/scripts/annotator/anchoring/types.coffee

```diff
@@ -0,0 +1,55 @@
+Annotator = require('annotator')
+$ = Annotator.$
+xpathRange = Annotator.Range
+
+
+# Helper function for throwing common errors
+missingParameter = (name) ->
+  throw new Error('missing required parameter "' + name + '"')
+
+
+###*
+# class:: RangeAnchor(range)
+#
+# This anchor type represents a DOM Range.
+#
+# :param Range range: A range describing the anchor.
+###
+class RangeAnchor
+  constructor: (root, range) ->
+    unless root? then missingParameter('root')
+    unless range? then missingParameter('range')
+    @root = root
+    @range = xpathRange.sniff(range).normalize(@root)
+
+  @fromRange: (root, range) ->
+    return new RangeAnchor(root, range)
+
+  # Create and anchor using the saved Range selector.
+  @fromSelector: (root, selector) ->
+    data = {
+      start: selector.startContainer
```

Comment:
Huh, you learn something every day. I didn't realise we'd diverged from the underlying implementation by renaming `{start, end}` to `{startContainer, endContainer}`.